diff --git a/proxy/proxy.go b/proxy/proxy.go
index db92051..54d36b4 100644
--- a/proxy/proxy.go
+++ b/proxy/proxy.go
@@ -504,7 +504,8 @@ func CopyRawConnIfExist(ctx context.Context, readerConn net.Conn, writerConn net
 				splice = false
 			}
 		}
-		if splice {
+		_ = splice
+		if false {
 			newError("CopyRawConn splice").WriteToLog(session.ExportIDToError(ctx))
 			statWriter, _ := writer.(*dispatcher.SizeStatWriter)
 			//runtime.Gosched() // necessary
