diff --color -ur sysprof-49.0/src/libsysprof-capture/sysprof-platform.c sysprof-49.0-patched/src/libsysprof-capture/sysprof-platform.c
--- sysprof-49.0/src/libsysprof-capture/sysprof-platform.c	2025-09-13 08:41:52.000000000 +0800
+++ sysprof-49.0-patched/src/libsysprof-capture/sysprof-platform.c	2026-01-13 14:53:57.945290385 +0800
@@ -60,6 +60,8 @@
 #include <stdlib.h>
 #include <sys/syscall.h>
 #include <unistd.h>
+#include <fcntl.h>
+#define O_BINARY 0
 
 #include "sysprof-capture-util-private.h"
 #include "sysprof-platform.h"
