diff --color -ur libfabric/src/common.c libfabric-patched/src/common.c
--- libfabric/src/common.c	2026-01-13 19:58:12.483193871 +0800
+++ libfabric-patched/src/common.c	2026-01-13 20:01:45.766796147 +0800
@@ -74,6 +74,13 @@
 #include <ofi_iov.h>
 #include <ofi_str.h>
 
+#ifndef SO_EE_ORIGIN_ZEROCOPY
+#define SO_EE_ORIGIN_ZEROCOPY 5
+#endif
+#ifndef SO_EE_CODE_ZEROCOPY_COPIED
+#define SO_EE_CODE_ZEROCOPY_COPIED 1
+#endif
+
 struct fi_provider core_prov = {
 	.name = "core",
 	.version = OFI_VERSION_DEF_PROV,
