diff --git a/phonopy/structure/cells.py b/phonopy/structure/cells.py
index 1271102c..2bbaec89 100644
--- a/phonopy/structure/cells.py
+++ b/phonopy/structure/cells.py
@@ -829,6 +829,9 @@ def get_cell_lines(cell: PhonopyAtoms, mapping=None, stars=None):
     magmoms = cell.magnetic_moments
     lattice = cell.cell
     lines = []
+    if stars is not None:
+        lines.append("Independent atoms (index start from 0):")
+        lines.append(", ".join([str(x) for x in stars]))
     lines.append("Lattice vectors:")
     lines.append("  a %20.15f %20.15f %20.15f" % tuple(lattice[0]))
     lines.append("  b %20.15f %20.15f %20.15f" % tuple(lattice[1]))
