diff --color -ur funasr_onnx-0.4.1/setup.py funasr_onnx-0.4.1-patched/setup.py
--- funasr_onnx-0.4.1/setup.py	2026-03-06 17:07:32.096187846 +0800
+++ funasr_onnx-0.4.1-patched/setup.py	2026-03-06 17:07:48.231123043 +0800
@@ -31,7 +31,7 @@
         "librosa",
         "onnxruntime>=1.7.0",
         "scipy",
-        "numpy<=1.26.4",
+        "numpy",
         "kaldi-native-fbank",
         "PyYAML>=5.1.2",
         "onnx",
