diff --git a/pyproject.toml b/pyproject.toml
index 6eddf93..21d75a2 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -53,6 +53,7 @@ where = ["./"]
 [tool.versioningit.vcs]
 default-tag = "0.0.1"
 method = "git"
+default-version = "0"
 
 [tool.mypy]
 ignore_missing_imports = true
diff --git a/pyproject.toml b/pyproject.toml
index 6eddf93..6e3939d 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -50,6 +50,9 @@ repository = "https://github.com/materialsproject/pymatgen-analysis-defects"
 exclude = ["tests"]
 where = ["./"]
 
+[tool.versioningit]
+default-version = "0"
+
 [tool.versioningit.vcs]
 default-tag = "0.0.1"
 method = "git"
